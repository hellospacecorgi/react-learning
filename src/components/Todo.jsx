import React from 'react';

export default function Todo({ todo }) {
    const { id, title, completed } = todo;
    const h1 = <h2>{title}</h2>
    const text = completed ? <strike>{h1}</strike> : h1;
    return <div data-testid="todo-1">{ text }</div>
}