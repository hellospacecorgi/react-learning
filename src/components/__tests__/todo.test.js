import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Todo from "../TodoList";
import renderer from "react-test-renderer";

//npm run test
test("sample test name", () => {
  expect(true).toBe(true);
});

// test('should render todo component completed false', () => {
//     const expectedTextContent = "wash dishes";
//     const testTodo = { id: 1, title:  expectedTextContent, completed: false, };
//     // render a component
//     render(<Todo todo={testTodo}/>);
//     // retrieve the component that is rendered
//     const todoElement = screen.getByTestId('todo-1');
//     expect(todoElement).toBeInTheDocument();
//     expect(todoElement).toHaveTextContent(expectedTextContent);
//     expect(todoElement).not.toContainHTML('strike');
// })

// test('should render todo component completed true', () => {
//     const expectedTextContent = "throw rubbish";
//     const testTodo = { id: 2, title:  expectedTextContent, completed: true, };
//     // render a component
//     render(<Todo todo={testTodo}/>);
//     // retrieve the component that is rendered
//     const todoElement = screen.getByTestId('todo-1');
//     expect(todoElement).toBeInTheDocument();
//     expect(todoElement).toHaveTextContent(expectedTextContent);
//     expect(todoElement).toContainHTML('strike');

//     expect(todoElement).toMatchSnapshot('<strike>');
// })

// test('should match snapshot', () => {
//     const expectedTextContent = "feed dog";
//     const testTodo = { id: 1, title:  expectedTextContent, completed: true, };

//     // render tree with component inside
//     const tree = renderer.create(<Todo todo={testTodo}/>);

//     expect(tree).toMatchSnapshot();
// })
