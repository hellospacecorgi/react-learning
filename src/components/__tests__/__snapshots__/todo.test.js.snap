// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h2>
      feed dog
    </h2>
  </strike>
</div>
`;

exports[`should render todo component completed true: <strike> 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h2>
      throw rubbish
    </h2>
  </strike>
</div>
`;
